src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJLibrary.java

```diff
@@ -0,0 +1,58 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.roots.OrderRootType;
+import com.intellij.openapi.roots.impl.libraries.LibraryEx;
+import com.intellij.openapi.roots.libraries.LibraryProperties;
+import com.intellij.openapi.roots.libraries.LibraryTable;
+import com.intellij.openapi.roots.libraries.PersistentLibraryKind;
+import fi.aalto.cs.apluscourses.model.Library;
+import java.nio.file.Path;
+import org.jetbrains.annotations.CalledWithWriteLock;
+import org.jetbrains.annotations.NotNull;
+
+public abstract class IntelliJLibrary<
```

Comment:
a nasty line wrap

Reply:
What would be the correct place? Written on one line it reads:
```java
public abstract class IntelliJLibrary<K extends PersistentLibraryKind<? extends LibraryProperties<S>>, S> extends Library {
//                                                                                                  ^ line wrap should be before this char
```
Perhaps just:
```java
public abstract class IntelliJLibrary
    <K extends PersistentLibraryKind<? extends LibraryProperties<S>>, S> extends Library {
```